### PR TITLE
Emoji image param

### LINKED ISSUE #1224: Make the ImageParameter support emojis
title says it all

### PR DESCRIPTION
**Pull request by metehus**
- Fixes #1224
  - Custom emoji as image
image
image

  - Default emoji as image
preview
    (also complex emojis)
a

- Quieres command fix
image

### CODE REVIEW

**Comment by pedrofracassi** [comment]
I was going to come here and suggest that you used Twemoji for regular emojis, but you read my mind. Stellar job.

File: index.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ try {
   require('canvas')
   require('./src/utils/CanvasUtils.js').initializeHelpers()
   canvasLoaded = true
-} catch (e) {}
+} catch (e) { }
```
[the remark this answers is not in the document]
Comment (metehus) [reply index.js:15]: blame VSCode code fix

**Comment by metehus** [comment]
btw lol #1234

File: src/utils/CanvasTemplates.js [hunk 2]
```diff
@@ -9,7 +9,7 @@ let CanvasUtils = {}
 try {
   Canvas = require('canvas')
   CanvasUtils = require('./CanvasUtils.js')
-} catch (e) {}
+} catch (e) { }
```
Comment (almeidx) [inline src/utils/CanvasTemplates.js:12]: 👀

**Review by almeidx — DISMISSED** [review]
LGTM

**Comment by almeidx** [comment]
Fix conflicts

**Review by Doges — DISMISSED** [review]
LGTM

File: index.js [hunk 3]
```diff
@@ -25,6 +25,10 @@ console.log(readFileSync('bigtitle.txt', 'utf8').toString())
 
 const Switchblade = require('./src/Switchblade.js')
 const client = new Switchblade(CLIENT_OPTIONS)
-client.on('debug', (...args) => console.log('debug', ...args))
-client.on('rateLimit', (...args) => console.log('rateLimit', ...args))
+
+if (process.env.NODE_ENV === 'PRODUCTION') {
+  client.on('debug', (...args) => console.log('debug', ...args))
+  client.on('rateLimit', (...args) => console.log('rateLimit', ...args))
+}
```
Comment (davipatury) [inline index.js:32]: why?

Comment (metehus) [reply index.js:32]: because logging debug logs on development isn't cool, it's too much spam and pretty hard to read the log

Comment (davipatury) [reply index.js:32]: but if it's a development instance it should print debug logs, I don't think it's too much spam tbh

Comment (metehus) [reply index.js:32]: fine

**Comment by almeidx** [comment]
conflicts

**Comment by Doges** [comment]
dijei rrrrrrrrzeitona

File: index.js [hunk 4]
```diff
@@ -25,6 +25,10 @@ console.log(readFileSync('bigtitle.txt', 'utf8').toString())
...
 const Switchblade = require('./src/Switchblade.js')
 const client = new Switchblade(CLIENT_OPTIONS)
-client.on('debug', (...args) => console.log('debug', ...args))
-client.on('rateLimit', (...args) => console.log('rateLimit', ...args))
+
+if (process.env.HIDE_DEBUG_LOGS !== 'TRUE') {
+  client.on('debug', (...args) => console.log('debug', ...args))
+  client.on('rateLimit', (...args) => console.log('rateLimit', ...args))
+}
+
```
Comment (fernando457829) [inline index.js:33]: It's already modified on #1247.

**Review by fernando457829 — CHANGES_REQUESTED** [review]

File: index.js (hunk 4 above)
Comment (metehus) [reply index.js:33]: pse amigo

**Review by fernando457829 — DISMISSED** [review]

**Review by Doges — APPROVED** [review]
É GALERA CHEGAMOS NA QUINTA FEIRA RAPAZEADA

**Pull request merged**